.pre-commit-config.yaml

```
@@ -0,0 +1,13 @@
+repos:
+- repo: local
+  hooks:
+  - id: linter
+    name: linter
+    entry: hatch run lint
+    language: system
+    types: [python]
+  - id: format
+    name: format
+    entry: hatch run format --diff
```

so commit will actually fail if there are formatting fixes needed
```suggestion
    entry: hatch run format --diff --exit-code
```